cirq/experiments/fidelity_estimation.py

```diff
@@ -227,3 +227,148 @@ def log_xeb_fidelity(
+
+
+def least_squares_xeb_fidelity_from_probabilities(
+        hilbert_space_dimension: int,
+        observed_probabilities: Sequence[Sequence[float]],
+        all_probabilities: Sequence[Sequence[float]],
+        observable_from_probability: Optional[Callable[[float], float]] = None,
+        normalize_probabilities: bool = True) -> Tuple[float, List[float]]:
+    """Least squares fidelity estimator with observable based on probabilities.
+
+    Using the notation from the docstring of
+    `least_squares_xeb_fidelity_from_expectations`, this function computes the
+    least squares fidelity estimate when the observable O_U has eigenvalue
+    corresponding to the computational basis state |z⟩ given by g(p(z)), where
+    p(z) = |⟨z|𝜓_U⟩|^2 and g is a function that can be specified. By default,
+    g is the identity function, but other choices, such as the logarithm, are
+    useful. By default, the probability p(z) is actually multiplied by the
+    Hilbert space dimension D, so that the observable is actually g(D * p(z)).
+    This behavior can be disabled by setting `normalize_probabilities` to
+    False.
+
+    Args:
+        hilbert_space_dimension: Dimension of the Hilbert space on which
+           the channel whose fidelity is being estimated is defined.
+        observed_probabilities: Ideal probabilities of bitstrings observed in
+            experiments. A list of lists, where each inner list contains the
+            probabilities for a single circuit.
+        all_probabilities: Ideal probabilities of all possible bitstrings.
+            A list of lists, where each inner list contains the probabilities
+            for a single circuit, and should have length equal to the Hilbert
+            space dimension. The order of the lists should correspond to that
+            of `observed_probabilities`.
+        observable_from_probability: Function that computes the observable from
+            a given probability.
+        normalize_probabilities: Whether to multiply the probabilities by the
+            Hilbert space dimension before computing the observable.
+
+    Returns:
+        A tuple of two values. The first value is the estimated fidelity.
+        The second value is a list of the residuals
+
+            f (e_U - u_U) - (m_U - u_U)
+
+        of the least squares minimization.
+    """
+    if not isinstance(observable_from_probability, np.ufunc):
+        if observable_from_probability is None:
+            observable_from_probability = lambda p: p
+        else:
+            observable_from_probability = np.frompyfunc(
+                observable_from_probability, 1, 1)
+    observable_from_probability = cast(Callable, observable_from_probability)
+    measured_expectations = []
+    exact_expectations = []
+    uniform_expectations = []
+    prefactor = hilbert_space_dimension if normalize_probabilities else 1.0
+    for observed, all_ in zip(observed_probabilities, all_probabilities):
```

Comment:
nit: I wonder whether we can come up with a better name for "all_". Maybe "observed" -> "observed_probs" and "all_" -> "all_probs"?

Reply:
Done.